File: apps/api/app/routers/ai_config.py
from fastapi import APIRouter, Depends

from app.schemas.ai_config import AiConfig
from app.schemas.response import ApiResponse, success_response
from app.services.ai_config_store import load_ai_config, save_ai_config
from app.services.auth_security import get_current_admin

router = APIRouter(tags=["admin-ai-config"], dependencies=[Depends(get_current_admin)])


@router.get("")
async def get_ai_config() -> ApiResponse[AiConfig]:
    """获取当前 OpenAI 兼容 API 配置。

    Returns:
        统一响应包裹的 AI 配置。
    """
    return success_response(data=load_ai_config())


@router.put("")
async def update_ai_config(payload: AiConfig) -> ApiResponse[AiConfig]:
    """保存 AI 配置。

    Args:
        payload: 完整 AI 配置请求体。

    Returns:
        保存后的 AI 配置。
    """
    saved = save_ai_config(payload)
    return success_response(data=saved)

File: apps/admin/src/api/ai-config.ts
import type { AiConfig } from "@/types/ai-config";
import { request, type RequestResult } from "@/lib/request";

/**
 * 获取当前 AI 配置。
 *
 * @returns OpenAI 兼容 API 配置
 */
export function fetchAiConfig(): Promise<RequestResult<AiConfig>> {
  return request<AiConfig>({ url: "/api/admin/ai-config", method: "GET" });
}

/**
 * 保存 AI 配置。
 *
 * @param payload - 完整 AI 配置
 * @returns 保存后的 AI 配置
 */
export function saveAiConfig(payload: AiConfig): Promise<RequestResult<AiConfig>> {
  return request<AiConfig>({
    url: "/api/admin/ai-config",
    method: "PUT",
    data: payload,
  });
}
